modules/gdscript/editor/gdscript_highlighter.cpp

```diff
@@ -495,7 +505,7 @@ Dictionary GDScriptSyntaxHighlighter::_get_line_syntax_highlighting_impl(int p_l
 					k++;
 				}
 
-				if (str[k] == ':') {
+				if (str[k] == ':' && !in_dict) {
```

Comment:
```suggestion
				if (str[k] == ':' && in_function_arg_dicts > 0) {
```

Reply:
Hi @MewPurPur ,

I think that condition on line 508 has to be inverted
e.g. i think any of the below would work
```
in_function_arg_dicts == 0 
in_function_arg_dicts <= 0
!in_function_arg_dicts
```